ush/forecast_postdet.sh

```diff
@@ -497,229 +469,198 @@ EOF
     done
   fi
 }
 
-FV3_GFS_nml(){
+FV3_nml(){
   # namelist output for a certain component
-  echo SUB ${FUNCNAME[0]}: Creating name lists and model configure file for FV3
+  echo SUB "${FUNCNAME[0]}": Creating name lists and model configure file for FV3
   # Call child scripts in current script directory
-  source $SCRIPTDIR/parsing_namelists_FV3.sh
+  source "${SCRIPTDIR}"/parsing_namelists_FV3.sh
   FV3_namelists
-  echo SUB ${FUNCNAME[0]}: FV3 name lists and model configure file created
-}
-
-DATM_nml(){
-  source $SCRIPTDIR/parsing_namelists_DATM.sh
-  DATM_namelists
-  echo SUB ${FUNCNAME[0]}: DATM name lists and model configure file created
+  echo SUB "${FUNCNAME[0]}": FV3 name lists and model configure file created
 }
 
-data_out_GFS() {
-  # data in take for FV3GFS
-  # Arguments: None
-  #
-  #------------------------------------------------------------------
-  # make symbolic links to write forecast files directly in memdir
+FV3_out() {
   echo "SUB ${FUNCNAME[0]}: copying output data for FV3"
-  #------------------------------------------------------------------
 
-  if [ $SEND = "YES" ]; then
-    # Copy model restart files
-    if [[ ${RUN} =~ "gdas" ]] && (( rst_invt1 > 0 )); then
-      cd $DATA/RESTART
-      mkdir -p "${COM_ATMOS_RESTART}"
-      for rst_int in $restart_interval ; do
-        if [ $rst_int -ge 0 ]; then
-          RDATE=$($NDATE +$rst_int $CDATE)
-          rPDY=$(echo $RDATE | cut -c1-8)
-          rcyc=$(echo $RDATE | cut -c9-10)
-          for file in "${rPDY}.${rcyc}0000."* ; do
-            ${NCP} "${file}" "${COM_ATMOS_RESTART}/${file}"
-          done
-        fi
+  # Copy FV3 restart files
+  if [[ ${RUN} =~ "gdas" ]]; then
+    cd "${DATA}/RESTART"
+    mkdir -p "${COM_ATMOS_RESTART}"
+    local idate=$(date -d "${current_cycle:0:8} ${current_cycle:8:2} + ${restart_interval} hours" +%Y%m%d%H)
+    while [[ ${idate} -le ${forecast_end_cycle} ]]; do
+      for file in "${idate:0:8}.${idate:8:2}0000."*; do
+        ${NCP} "${file}" "${COM_ATMOS_RESTART}/${file}"
       done
-      if [ $DOIAU = "YES" ] || [ $DOIAU_coldstart = "YES" ]; then
-        # if IAU is on, save restart at start of IAU window
-        rst_iau=$(( ${IAU_OFFSET} - (${IAU_DELTHRS}/2) ))
-        if [ $rst_iau -lt 0 ];then
-          rst_iau=$(( (${IAU_DELTHRS}) - ${IAU_OFFSET} ))
-        fi
-        RDATE=$($NDATE +$rst_iau $CDATE)
-        rPDY=$(echo $RDATE | cut -c1-8)
-        rcyc=$(echo $RDATE | cut -c9-10)
-        for file in "${rPDY}.${rcyc}0000."* ; do
-          ${NCP} "${file}" "${COM_ATMOS_RESTART}/${file}"
-        done
-      fi
-    elif [[ ${RUN} =~ "gfs" ]]; then
-      ${NCP} "${DATA}/input.nml" "${COM_ATMOS_HISTORY}/input.nml"
-    fi
+      local idate=$(date -d "${idate:0:8} ${idate:8:2} + ${restart_interval} hours" +%Y%m%d%H)
+    done
+  elif [[ ${RUN} =~ "gfs" ]]; then
+    # No need to copy FV3 restart files when RUN=gfs
```

Comment:
NB: Right now we are linking the entire restart directory to COM for gfs (`FV3_predet()` L189)

Reply:
Ok. Will revisit later.